core/src/main/java/org/aerogear/mobile/core/metrics/Metrics.java

```diff
@@ -1,11 +1,9 @@
 package org.aerogear.mobile.core.metrics;
 
-import java.util.Map;
-
 public interface Metrics {
 
     String identifier();
 
-    Map<String, String> data();
+    Object data();
```

Comment:
Had to do this to support `JSONArray`. `JSONArray` and `JSONObject` don't share any ancestors.

The goal is to be able to have an entry with the shape: 
```json
{
  {{identifier}}: [data]
}
```

Reply:
Both implementations we provide use JSONObject, so why not a JSONObject?  In the future if we need this then we can raise the interface's type without breaking compatibility.

Alternatively we can define the the Metrics data as a generic type.

Reply:
We want to reach the following shape from https://github.com/aerogear/aerogear-app-metrics/pull/33:
```
"security": [
      {
        "type": "org.aerogear.mobile.security.checks.DeveloperModeCheck",
        "passed": true
      },
     {
        "type": "org.aerogear.mobile.security.checks.EmulatorCheck",
        "passed": false
      },
      {
        "type": "org.aerogear.mobile.security.checks.DebuggerCheck",
        "passed": false
      },
      {
        "type": "org.aerogear.mobile.security.checks.RootedCheck",
        "passed": false
      },
      {
        "type": "org.aerogear.mobile.security.checks.ScreenLockCheck",
        "passed": false
      }
    ]
```

So I changed `SecurityCheckResultMetric` to be `{identifier: "security", data: JSONArray()}`.
I guess we could somehow change `NetworkMetricsPublisher#publish` instead.

Do you see any benefit of having `Metrics<T>` without a type constraint over `Object`?

I didn't see much point doing that because even https://developer.android.com/reference/org/json/JSONObject.html#put(java.lang.String,java.lang.Object) receives `Object` over having overloads for `JSONObject|Array`

Reply:
The best thing that a generic type parameter gets us is a little bit of developer ease at code writing time.  Depending on how much people will put in their Metrics implementations it may or may not be beneficial.

What the generic type does is make explicit the data type that is being wrapped by the metrics class.  Functionality wise it won't behave differently than the current proposal, but it does make the down typing you can do on the data method more explicit.